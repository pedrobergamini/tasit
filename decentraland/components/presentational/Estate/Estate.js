import React from "react";
import { Image, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import {
  responsiveWidth,
  responsiveHeight,
} from "react-native-responsive-dimensions";

export default function Estate({ estate }) {
  let { img } = estate;

  return (
    <React.Fragment>
      <Image style={styles.landImage} source={{ uri: img }} />
    </React.Fragment>
  );
}

Estate.propTypes = {
  estate: PropTypes.object.isRequired,
};

const styles = StyleSheet.create({
  landImage: {
    height: responsiveHeight(20),
    width: responsiveWidth(95),
  },
});
