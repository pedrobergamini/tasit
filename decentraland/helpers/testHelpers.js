import AssetTypes from "@constants/AssetTypes";
const { ESTATE, PARCEL } = AssetTypes;

export const parcel = {
  type: PARCEL,
  id: "0123456789",
  name: "Sample Parcel",
  img: "https://api.decentraland.org/v1/parcels/0/0/map.png",
};

export const estate = {
  type: ESTATE,
  id: "123",
  name: "Sample Estate",
  img: "https://api.decentraland.org/v1/estates/1/map.png",
};

export const estateForSale = {
  id: "1",
  priceMana: 0,
  priceUSD: 0,
  asset: estate,
};

export const parcelForSale = {
  id: "2",
  priceMana: 0,
  priceUSD: 0,
  asset: parcel,
};

export const anAction = {
  getId: async () =>
    "0x1234567890123456789012345678901234567890123456789012345678901234",
};

export default {
  estateForSale,
  parcelForSale,
  parcel,
  estate,
  anAction,
};
