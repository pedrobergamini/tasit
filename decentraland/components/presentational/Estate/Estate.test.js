import React from "react";
import { shallow } from "enzyme";
import Estate from "@presentational/Estate";
import { estate } from "@helpers/testHelpers";

describe("Estate", () => {
  it("renders the component", async () => {
    expect(shallow(<Estate estate={estate} />)).toMatchSnapshot();
  });
});
