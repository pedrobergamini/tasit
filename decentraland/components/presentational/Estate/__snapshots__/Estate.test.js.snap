// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estate renders the component 1`] = `
<Fragment>
  <Component
    source={
      Object {
        "uri": "https://api.decentraland.org/v1/estates/1/map.png",
      }
    }
    style={
      Object {
        "height": 266.8,
        "width": 712.5,
      }
    }
  />
</Fragment>
`;
